web/modules/custom/dbcdk_community/src/Form/QuarantineCreateForm.php

```diff
@@ -0,0 +1,177 @@
+      $profile = $profile_api->profileFindOne(json_encode($profile_filter));
+    }
+    catch (ApiException $e) {
+      \Drupal::logger('DBCDK Community Service')->error($e);
+      // Create an empty Profile instance. The constructor expects a Profile
+      // object and will cause a fatal error if it receives something else and
+      // we would like to display an empty page with a Drupal message in case of
+      // an error.
+      $profile = new Profile();
+    }
+
+    return new static(
+      $profile,
+      $container->get('dbcdk_community.api.quarantine')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'dbcdk_community_quarantine_create_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    // Defensive coding to make sure we only display the "create quarantine"
+    // form if we could find a Community Profile.
+    if (empty($this->profile->getUsername())) {
+      drupal_set_message($this->t('The system could not find the profile that should receive this quarantine. Please try again later or contact an administrator.'), 'error');
+      return $form;
+    }
+
+    $form['reason'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Reason'),
+      '#required' => TRUE,
+    ];
+
+    $form['start'] = [
+      '#type' => 'date',
+      '#title' => $this->t('Start date'),
+      '#date_date_format' => self::DATE_FORMAT,
+      '#default_value' => date(self::DATE_FORMAT),
+      '#required' => TRUE,
+    ];
+
+    $form['end'] = [
+      '#type' => 'date',
+      '#title' => $this->t('End date'),
+      '#date_date_format' => self::DATE_FORMAT,
+      '#default_value' => date(self::DATE_FORMAT),
+      '#required' => TRUE,
+    ];
+
+    $form['actions']['#type'] = 'actions';
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Create'),
+      '#button_type' => 'primary',
+    ];
+    $form['actions']['cancel'] = [
+      '#type' => 'link',
+      '#title' => $this->t('Cancel'),
+      '#url' => Url::fromRoute('page_manager.page_view_dbcdk_community_profile', [
+        'username' => $this->profile->getUsername(),
+      ]),
+      '#attributes' => ['class' => ['button']],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    try {
+      // Update fields on the Community Profile. The elements in the $fields
+      // array represents the fields we wish to update.
+      $fields = [
+        'reason',
+        'start',
+        'end',
+      ];
+      $quarantine = [];
+      foreach ($fields as $field) {
+        $quarantine[$field] = $form_state->getValue($field);
+      }
+
+      // Set the ID of the Drupal Moderator who created this quarantine.
+      $quarantine['creatorId'] = $this->currentUser()->id();
```

Comment:
I think it would be good to have a little more elaborate description of why we set information about the Drupal user here and not a Community service id.